great_expectations/datasource/fluent/sources.py

```diff
@@ -271,8 +271,27 @@ def _bind_asset_factory_method_if_not_present(
             def _add_asset_factory(
                 self: Datasource, name: str, **kwargs
             ) -> pydantic.BaseModel:
+
+                # if the data_connector uses a data_connector we need to identify the
+                # asset level attributes needed by the data_connector
+                # push them to `dc_options` field
+                if self.data_connector_type:
+                    dc_options = {
```

Comment:
You could combine these loops:
```
dc_options = {}
for k, v in kwarg.items():
  if k in self.data_connector_type.asset_level_option_keys and v:
    dc_options[k] = kwargs.pop(k)
if dc_options:
  self.data_connector_type.asset_options_type(**dc_options)
```

Reply:
I did this initially but I can't `pop()` while iterating over kwargs.
I'd have to make a copy, which I could do.